djangocms_blog/models.py

```diff
@@ -51,6 +51,23 @@ class KnockerModel:
         pass
 
 
+def _get_language(instance, language):
+    available_languages = instance.get_available_languages()
+    if language and language in available_languages:
+        return language
+    language = get_language()
+    if language and language in available_languages:
+        return language
+    language = instance.get_current_language()
+    if language and language in available_languages:
+        return language
+    if get_setting('USE_FALLBACK_LANGUAGE_IN_URL'):
+        fallback_languages = instance.get_fallback_languages()
```

Comment:
I have not checked parler code, but IIRC `get_fallback_languages` does not check languages against the ones available on the current instance. Maybe we should filter `fallback_languages` against `available_languages` to pick an existing one

Reply:
I checked it and that's indeed the case. I pushed another revision.